pretrain/tinyllama.py

```diff
@@ -0,0 +1,340 @@
+"""
+This script is adapted from TinyLlama:
+https://github.com/jzhang38/TinyLlama/blob/main/pretrain/tinyllama.py
+"""
+import math
+import sys
+import time
+from functools import partial
+from pathlib import Path
+from typing import Tuple, Union
+
+import lightning as L
+import torch
+import torch.nn as nn
+from lightning.data import StreamingDataset
+from lightning.data.streaming.item_loader import TokensLoader
+from lightning.fabric.strategies import FSDPStrategy
+from lightning.fabric.utilities.throughput import Throughput, get_available_flops, measure_flops
+from lightning.fabric.loggers import CSVLogger
+from lightning.pytorch.loggers import WandbLogger
+from torch.utils.data import DataLoader
+
+# support running without installing as a package
+wd = Path(__file__).parent.parent.resolve()
+sys.path.append(str(wd))
+
+from lit_gpt.model import GPT, Block, Config, LLaMAMLP
+from lit_gpt.packed_dataset import CombinedDataset
+from lit_gpt.utils import chunked_cross_entropy, num_parameters
+
+# System settings
+model_name = "tiny-llama-1.1b"
+name = "lit-tiny-llama-1.1b"
```

Comment:
`name` is used only here once, when defining `out_dir`. When I first read the code I expected that this variable will be used later on, but nope. Creates an ambiguity.

Reply:
This is so that I can change the name easily for different experiments, to avoid cluttering the same output dir with checkpoints.